controller/celebrant.controller.js

```diff
@@ -0,0 +1,253 @@
+const {pool} = require('../config/dbConnect');
+const { validationResult } = require('express-validator');
+const createCelebrantValidation = require('../utils/validator');
+
+
+//create a new celebrant
+const createCelebrant = async(req, res) => {
+
+     // Check for validation errors
+
+     const errors = validationResult(req);
+
+
+     if (!errors.isEmpty()) {
+         return res.status(400).json({ success: false, errors: errors.array() });
+     }
+
+    const {username,gender,phone_number,email,birthdate,channel,is_active} = req.body
+
+    try {
+         //check if celebrant exists
+         const celebrantExistsQuery = {text: "SELECT * FROM celebration.celebrants WHERE username = $1",
+         values: [username]
+        };
+
+        const { rows } = await pool.query(celebrantExistsQuery);
+
+        if (rows.length > 0) {
+            const celebrant = rows[0]
+
+        }
+
+        //insert new celebrant
+        const result = {
+            text: "INSERT INTO celebration.celebrants (username,gender,phone_number,email,birthdate,channel,is_active) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *",
+            values: [username,gender,phone_number,email,birthdate,channel,true]
+        }
+
+        const {rows: newCelebrant}= await pool.query(result);
```

Comment:
`newCelebrant` will make anyone reading your code think that it is an object. `rows` specifically means more than one row, which means that it is an array.

Use a more descriptive variable name.

Reply:
sorted sir

thanks